SuggestedNameAndDir.md

```diff
@@ -0,0 +1,240 @@
+    description: 'Markdown',
+    accept: {
+      'text/markdown': ['.md'],
+    },
+  }],
+});
+```
+
+### Specifying starting directory based on an existing handle.
+
+There are two cases where it is useful to be able to specify a starting
+directory based on an existing handle. The first of these if a website wants to
+let a user pick a sibling to an existing file or directory. For this purpose we
+propose adding a `startInParentOf` option, that can be passed to any of the file
+and directory picker methods:
+
+```javascript
+const existing_handle = /* some FileSystemHandle, could be either a file or a directory */;
+const file_ref = await self.showSaveFilePicker({
+  startInParentOf: existing_handle
+});
+```
+
+The second situation is where a website wants to prompt the user to open a file
+in a directory it already has a handle to. For example if a website has a
+"project" directory open, it might make sense for a "Save" dialog to start out
+in that directory. For this purpose we propose additionally adding a `startIn`
+option:
+
+```javascript
+const existing_dir_handle = /* some FileSystemDirectoryHandle */;
+const file_ref = await self.showSaveFilePicker({
+  startIn: existing_dir_handle
+});
+```
+
+### Specifying a well-known starting directory
+
+To support saving files in or opening files from certain well-known directories,
+we also propose allowing passing certain string values to `startIn` to represent
+these well-known directories. This would look something like:
+
+```javascript
+const file_ref = await self.showOpenFilePicker({
+  startIn: 'pictures'
+});
+```
+
+The possible values for `startIn` would be:
+
+- `desktop`: The user's Desktop directory, if such a thing exists.
+- `documents`: Directory in which documents created by the user would typically be stored.
+- `downloads`: Directory where downloaded files would typically be stored.
+- `home`: The user's Home directory.
+- `music`: Directory where audio files would typically be stored.
+- `pictures`: Directory where photos and other still images would typically be stored.
+- `videos`: Directory where videos would typically be stored.
+
+### Distinguishing the "purpose" of different file picker invocations.
+
+Currently the Chrome implementation of the File System Access API remembers the
+last used directory on a per-origin basis. To allow websites to remember
+different directories for file pickers that are used for different purposes
+(i.e. opening documents, exporting to other formats, inserting images) we
+propose adding an `id` option to the file and directory picker methods. If an
+`id` is specified the file picker implementation will remember a separate last
+used directory for pickers with that same `id`:
+
+```javascript
+const file_ref1 = await self.showSaveFilePicker({
+  id: 'fruit'
+});
+
+const file_ref2 = await self.showSaveFilePicker({
+  id: 'flower'
+});
+```
+
+## Detailed design discussion
+
+### Interaction of `suggestedName` and accepted file types
```

Comment:
Should we discuss behavior in situations where dangerous extensions are suggested? Ex: `suggestedName: 'file.lnk'`

Reply:
I added a security considerations section mentioning some of this.